test/hotspot/jtreg/compiler/intrinsics/math/TestMinMaxInlining.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2024, Red Hat, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8307513
+ * @summary Test min and max IR inlining decisions
+ * @library /test/lib /
+ * @run driver compiler.intrinsics.math.TestMinMaxInlining
+ */
+
+package compiler.intrinsics.math;
+
+import compiler.lib.ir_framework.Argument;
+import compiler.lib.ir_framework.Arguments;
+import compiler.lib.ir_framework.Check;
+import compiler.lib.ir_framework.CompilePhase;
+import compiler.lib.ir_framework.IR;
+import compiler.lib.ir_framework.IRNode;
+import compiler.lib.ir_framework.Test;
+import compiler.lib.ir_framework.TestFramework;
+
+public class TestMinMaxInlining {
+    public static void main(String[] args) {
+        TestFramework.run();
+    }
+
+    @Test
+    @Arguments(values = { Argument.NUMBER_MINUS_42, Argument.NUMBER_42 })
+    @IR(counts = { IRNode.MIN_I, "1" })
+    private static int testIntMin(int a, int b) {
+        return Math.min(a, b);
+    }
+
+    @Check(test = "testIntMin")
+    public static void checkTestIntMin(int result) {
+        if (result != -42) {
+            throw new RuntimeException("Incorrect result: " + result);
+        }
+    }
+
+    @Test
+    @Arguments(values = { Argument.NUMBER_MINUS_42, Argument.NUMBER_42 })
+    @IR(counts = { IRNode.MAX_I, "1" })
+    private static int testIntMax(int a, int b) {
+        return Math.max(a, b);
+    }
+
+    @Check(test = "testIntMax")
+    public static void checkTestIntMax(int result) {
+        if (result != 42) {
+            throw new RuntimeException("Incorrect result: " + result);
+        }
+    }
+
+    @Test
+    @Arguments(values = { Argument.NUMBER_MINUS_42, Argument.NUMBER_42 })
+    @IR(phase = { CompilePhase.BEFORE_MACRO_EXPANSION }, counts = { IRNode.MIN_L, "1" })
+    @IR(phase = { CompilePhase.AFTER_MACRO_EXPANSION }, counts = { IRNode.MIN_L, "0" })
+    private static long testLongMin(long a, long b) {
```

Comment:
Can you add a comment why it disappears after macro expansion?

Reply:
Good question. On non-avx512 machines after macro expansion the min/max nodes become cmov nodes, but but that's not the full story because on avx512 machines, they become minV/maxV nodes. Would you tweak the `@IR` annotations to capture this? Or would you leave it just as a comment?